### ISSUE TITLE
C2D Flow

### ISSUE THREAD

**Issue by AnaLoznianu** [oceanprotocol/ocean.py#619]
- [x] Implement `test_compute_registered_dataset` (#630)
- Fix TODOs introduced in 630. See for details: https://github.com/oceanprotocol/ocean.py/pull/630#issue-1093777094
    - [x] Reinstate type enforcement on `DataServiceProvider.start_compute_job` - Unable to fix, so left disabled
    - [x] Fix docstring for `DataServiceProvider.start_compute_job` (what is algorithm_custom_data)?
    - [x] Refactor compute flow tests to use high-level `OceanCompute` and `OceanAssets` APIs instead of `DataServiceProvider` and `ERC20Token` APIs.
    - [x] Pass valid compute environment ids to `initialize` and `startCompute` endpoints - Won't Fix until Provider supports compute environment
- [x] Reinstate `test_compute_raw_algo`
- [x] Reinstate `test_compute_multi_inputs`
- [x] Reinstate `test_update_trusted_algorithms` (blocked by https://github.com/oceanprotocol/ocean.py/pull/646)
- [x] Reinstate `test_compute_trusted_algorithms`
- [x] Review `test_ocean_compute`.  Tests are currently passing, but we suspect updates are required to bring them up to date for v4

**Comment by DMats** [comment]
@alexcos20 was able to successfully start a v4 compute job in https://github.com/oceanprotocol/ocean.js/pull/1196.  I'll use this as a reference while implementing v4 compute flow in ocean.py

**Comment by DMats** [comment]
Today, I was able to successfully start a compute job and view the status.  I have not yet been able to successfully stop the compute job.
